proto/slinky/mm2/v1/query.proto

```diff
@@ -0,0 +1,85 @@
+syntax = "proto3";
+package slinky.mm2.v1;
+
+import "gogoproto/gogo.proto";
+import "google/api/annotations.proto";
+import "slinky/types/v1/currency_pair.proto";
+import "slinky/mm2/v1/market.proto";
+import "slinky/mm2/v1/params.proto";
+
+option go_package = "github.com/skip-mev/slinky/x/mm2/types";
+
+// Query is the query service for the x/marketmap module.
+service Query {
+  // MarketMap returns the full market map stored in the x/marketmap
+  // module.
+  rpc MarketMap(MarketMapRequest) returns (MarketMapResponse) {
+    option (google.api.http).get = "/slinky/mm2/v1/marketmap";
+  }
+
+  // Market returns a market stored in the x/marketmap
+  // module.
+  rpc Market(MarketRequest) returns (MarketResponse) {
+    option (google.api.http).get = "/slinky/mm2/v1/market";
+  }
+
+  // LastUpdated returns the last height the market map was updated at.
+  rpc LastUpdated(LastUpdatedRequest) returns (LastUpdatedResponse) {
```

Comment:
why do we need a separate query for last updated when marketmap query exists?

Reply:
This query has significantly lower state read overhead (O(1)). 

To get the market map from state, it is O(N).

Clients can be configured to only poll the `LastUpdated` field, and if it is different from some value they have cached, only then do they need the full query. 

Just a potential optimization.